force-app/main/default/classes/ClassesObjectsMethods.cls

```diff
@@ -126,33 +147,41 @@ public with sharing class ClassesObjectsMethods {
         List<String> scientistNames = new List<String>{'Tim Berners-Lee', 'Alan Turing', 'Grace Hopper', 'Donald Knuth', 'Guido van Rossum', 'Ken Thompson', 'Stephen Hawking'};
 
         // The variable to store the concatenated string
+        String scientistString = '';
 
         // Loop through the list of scientists' names
-
-            // If the name doesn't contain the letter 'G', skip this iteration
-
+        for (String name : scientistNames) {
             // Add the name to the result string, followed by a comma
-
+            if (name.contains('g') || name.contains('G')) {
+                scientistString += name + ',';
+            }
+        }
         // Remove the last comma from the result string
-
-        return null; // Replace null with the variable you used to store the result
+        if(scientistString.endsWith(',')) {
+            scientistString = scientistString.substring(0, scientistString.length() - 1);
+        }
+        return scientistString;
```

Comment:
Good work.  Rather than doing the OR condition I  for case sensitivity you could  use the ignoreCase mathod or call toLower or toUpper on everything. 


check out the remove end method. It would have saved you some work.